### PR TITLE
Enable Universal Profiling as Enterprise feature

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we ensure that Universal Profiling can only be used with an Enterprise license.

### CODE REVIEW

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingPlugin.java [hunk 1]
```diff
@@ -108,10 +111,12 @@ public Collection<Object> createComponents(
             registry.get().initialize();
             indexManager.get().initialize();
             dataStreamManager.get().initialize();
-            return List.of(registry.get(), indexManager.get(), dataStreamManager.get());
-        } else {
-            return Collections.emptyList();
         }
+        return Collections.singletonList(createLicenseChecker());
```
Comment (danielmitterdorfer) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingPlugin.java:115]: See https://github.com/elastic/elasticsearch/pull/97546#discussion_r1339564573 for context on why we are returning the license checker in all cases (but no other components).

**Comment by danielmitterdorfer** [comment]
`elasticsearch-ci/part-1` failed because of https://github.com/elastic/elasticsearch/issues/100251 

@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by danielmitterdorfer** [comment]
BWC tests were flaky.

@elasticsearchmachine run elasticsearch-ci/bwc

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

**Comment by rockdaboot** [comment]
The curl command works as expected together with your ES startup (fresh data).

But when starting ES with `-Drun.license_type=trial --preserve-data`, then restart ES with `-Drun.license_type=basic --preserve-data`, the new license is not recognized (curl gives 200 OK).

Also the other way round, first using basic and then trial, leaves me with a 403 response.

I guess this behavior has nothing to do with this PR, but want to make sure that this is expected.

**Comment by danielmitterdorfer** [comment]
> But when starting ES with -Drun.license_type=trial --preserve-data, then restart ES with -Drun.license_type=basic --preserve-data, the new license is not recognized (curl gives 200 OK).

Ah, good catch but that's expected. The license is preserved in the cluster state and it's not that you can change the license at will with a dev cluster using the `run.license_type` system property. Instead you'd need to use the [licensing APIs](https://www.elastic.co/guide/en/elasticsearch/reference/current/licensing-apis.html) for this.

**Comment by danielmitterdorfer** [comment]
Also, I'm gonna look into these failing CI checks. Unfortunately most of the times it's just caused by unrelated flaky tests but I'll investigate these.

**Review by rockdaboot — APPROVED** [review]
LGTM, testing the PR works as expected.

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/part-3

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

**Comment by danielmitterdorfer** [comment]
Merging `main` after I've muted several failing tests in https://github.com/elastic/elasticsearch/pull/100384 and https://github.com/elastic/elasticsearch/pull/100385.

@elasticmachine merge upstream

**Comment by danielmitterdorfer** [comment]
@elasticmachine merge upstream

**Comment by danielmitterdorfer** [comment]
build timed out (CI infrastructure issue)

@elasticsearchmachine run elasticsearch-ci/part-1

**Pull request merged**